### DISCUSSION TITLE
Pivot Columns are not returned with Adonis 5

### DISCUSSION THREAD

**Discussion by mohux** [adonisjs/core#1340] [Help]
I really spent hours trying to debug the issue, please help

I have an "orders" table and model
a "services" table and model

they have many to many relationship

i added this

![image](https://user-images.githubusercontent.com/57658626/88487334-d12d4f00-cf8c-11ea-839a-226700ce9eca.png)

when I run this code 
![image](https://user-images.githubusercontent.com/57658626/88487338-ea360000-cf8c-11ea-94fb-94bd52b100a2.png)
the pivot_price is not returned, even though when I run the same generated select query in PSQL, and it works perfectly

Is there anything am doing wrong that is preventing the model from returning the pivot columns?

**Comment by mohux** [comment]
Though Database is working perfectly as well but am losing the advantage of Lucid if I rely too much on Database :(

**Comment by thetutlage** [comment]
How are you trying to access the values?

**Reply by mohux** [reply]
the pivot values? 

I tried this and it returns undefined 
![image](https://user-images.githubusercontent.com/57658626/88487503-477e8100-cf8e-11ea-9268-a1b11f83496d.png)
and am trying to return it with the query

**Reply by mohux** [reply]
![image](https://user-images.githubusercontent.com/57658626/88487509-5cf3ab00-cf8e-11ea-972c-a3109c7bc681.png)

Just to clarify, saloon has  services and orders, orders are related to services and a user who ordered something from saloon 
just dummy select it wont be in production like this

but what is not working is getting pivot_price

**Reply by thetutlage** [reply]
Can you please atleast type your code inline here vs dumping screenshots?

And it will be nice you can share the complete code to avoid back and forth.

- in which variable you store the query results.
- how do you access the computed property from that variable?

**Reply by mohux** [reply]
I apologise i thought the screenshots are good, i will paste them now

I tried the simplest query like below 

``` 
      const order = await Order.query()
        .where("saloon_id", params.id)
        .preload("services", (query) => query.pivotColumns(["price"]))
        .first();

      return order?.pivot_price;
```
the computed is inside order

```

 @manyToMany(() => Service, {
    pivotTable: "order_services",
    localKey: "id",
    relatedKey: "id",
    pivotForeignKey: "order_id",
    pivotRelatedForeignKey: "service_id",
    pivotColumns: ["price"],
  })
  public services: ManyToMany<typeof Service>;

  @computed()
  public get pivot_price() {
    return this.$extras.pivot_price;
  }
```

**Reply by mohux** [reply]
Ok now I feel dumb 😮  your question gave me the answer, the computed should've been inside services not inside order (since its the one that is getting inner joined with pivot table)

**Comment by thetutlage** [comment]
Nice and screenshots are not okay when viewing on mobile.

See the difference yourself 
![Screenshot_20200727-010314_Chrome](https://user-images.githubusercontent.com/1706381/88487724-3db44800-cfa5-11ea-8c73-cbc4f3cc8044.jpg)

![Screenshot_20200727-010321_Chrome](https://user-images.githubusercontent.com/1706381/88487733-473db000-cfa5-11ea-9430-6c8af7840bb7.jpg)

**Reply by mohux** [reply]
>.< am truely sorry, and thanks for your fast replay I really appreciate it I was thinking way out of the box that i forgot about the basics
